pkg/client/rootd/dns/server.go

```diff
@@ -249,6 +260,24 @@ func (s *Server) resolveInCluster(c context.Context, q *dns.Question) (result dn
 	return result, rCode, nil
 }
 
+func (s *Server) ResolveAlias(query string) *string {
+	for i := range s.config.Mappings {
+		mappingName := s.config.Mappings[i].Name + "."
+		if strings.Count(mappingName, ".") == 1 {
+			mappingName += s.clusterDomain
+		}
```

Comment:
What use-case does this cover. The name just before the cluster domain is almost always "svc".

Reply:
When I receive a query for let's say "echo-service", then `query` = 'echo-service.cluster.local' since it's using the default resolver, so to make it "match" to the mapping, I need to add the domain. Do I miss something ?

Reply:
"echo-service" is most likely a reference to `echo-service.<namespace>.svc.<clusterDomain>`. It's never `echo-service.<clusterDomain>`, so it's wrong in two ways.
1. We must not assume that the mapping includes a namespace.
2. The `svc` part is missing regardless.